src/hacl/hkdf.rs

```diff
@@ -0,0 +1,150 @@
+                        info.len()
+                            .try_into()
+                            .map_err(|_| Error::ArgumentsTooLarge)?,
+                        OKM_LEN.try_into().map_err(|_| Error::ArgumentsTooLarge)?,
+                    );
+                }
+                Ok(okm)
+            }
+
+            /// HKDF using the `salt`, input key material `ikm`, `info`. The output length
+            /// is defined through the result type.
+            /// Calls `extract` and `expand` with the given input.
+            ///
+            /// Returns the key material in an array of length `okm_len`.
+            pub fn hkdf<const OKM_LEN: usize>(
+                salt: &[u8],
+                ikm: &[u8],
+                info: &[u8],
+            ) -> Result<[u8; OKM_LEN], Error> {
+                let prk = extract(salt, ikm);
+                expand(&prk, info)
+            }
+
+            /// This module uses heap allocated vectors for cases where the output
+            /// length is not const.
+            pub mod vec {
+                use super::super::Error;
+
+                /// HKDF expand using the pre-key material `prk` and `info`. The output length
+                /// is defined through the result type.
+                /// Returns the key material in an array of length `okm_len` or
+                /// [`Error::OkmTooLarge`] if the requested `okm_len` is too large.
+                ///
+                /// Note that this function returns an [`Error::ArgumentsTooLarge`]
+                /// if `salt`, `ikm`, or `OKM_LEN` is larger than 2**32 bytes.
+                pub fn expand(prk: &[u8], info: &[u8], okm_len: usize) -> Result<Vec<u8>, Error> {
+                    if okm_len > 255 * $tag_len {
+                        // Output size is too large. HACL doesn't catch this.
+                        return Err(Error::OkmTooLarge);
+                    }
+                    let mut okm = vec![0u8; okm_len];
+                    unsafe {
+                        libcrux_hacl::$expand(
+                            okm.as_mut_ptr(),
+                            prk.as_ptr() as _,
+                            prk.len().try_into().map_err(|_| Error::ArgumentsTooLarge)?,
+                            info.as_ptr() as _,
+                            info.len()
+                                .try_into()
+                                .map_err(|_| Error::ArgumentsTooLarge)?,
+                            okm_len.try_into().map_err(|_| Error::ArgumentsTooLarge)?,
+                        );
+                    }
+                    Ok(okm)
+                }
+
+                /// HKDF using the `salt`, input key material `ikm`, `info`. The output length
+                /// is defined through the result type.
+                /// Calls `extract` and `expand` with the given input.
+                ///
+                /// Returns the key material in an array of length `okm_len`.
+                pub fn hkdf(
```

Comment:
This function is unused because it's implemented on the top level. Lets remove it.

Reply:
Removed!